### ISSUE TITLE
Building V on Haiku

### ISSUE THREAD

**Issue by ttcoder** [vlang/v#10435]
**TL;DR** :

When compiling V from source, it seems V mistakenly detects the Haiku platform by looking for it with `"#ifdef __haiku__" ` (lowercase) whereas it should be using the uppercase variant, i.e. `#ifdef "__HAIKU__"`

The culprit seems to be in comptime.v, the `"return __haiku__"` line, but I'm not sure.

**Long version** :

Just found out about this language, I'm loving what I see so far!
Managed to build V for my platform (Haiku) but it took a bit of digging:

When running "make" on the out-of-the-box V source, the resulting "v" won't run correctly, it produces a panic() about "unknown OS". So what I did was to download the vc/v.c by hand, compile it, play with the defines. I tried a few variants, replacing all "haiku" with "HAIKU", then replacing only part of them... In the end I ended up calling `cc... -DOpenBSD ...` (!) and the resulting executable is able to run fibonacci.v and so on, so I'm happilly testing various V programs now.

Since there is already a bunch of "haiku" references in the source code, I take it to mean you guys are ok with supporting it, so it makes sense for me to submit the one-liner fix above. Thank you.

**Comment by medvednikov** [comment]
Of course :)

Strange, I'm pretty sure it used to compile. Maybe they changed their ifdef.

**Comment by medvednikov** [comment]
@ttcoder should be fixed, can you try again?

Just had to use an all cap ifdef.

**Comment by ttcoder** [comment]
Much smoother experience, bolshoi spasibo ^-^
Just had to rem out the `#include <sys/ptrace.h>` statement and then V compiles and runs the basic examples. Seems to compile faster than a couple weeks ago too.

I'll wait until I upgrade to Haiku R1/beta3 (I'm currently still on /beta1) to submit patches (for e.g. networking to function), in case the latest Haiku is functionally different. Keep rocking !

**Comment by medvednikov** [comment]
> Just had to rem out the #include

you can try doing 

```
$if !haiku {
#include ...
```

if that works, please submit a PR :)
